### PR TITLE
Implement Prism -> Sorbet translation for interpolated matches

### LINKED ISSUE #117: Implement translation for `PM_INTERPOLATED_MATCH_LAST_LINE_NODE`

### PR DESCRIPTION
**Pull request by amomchilov**
Similar to #288, but for interpolated regexes. Closes #117.

### CODE REVIEW

File: parser/prism/Translator.cc [hunk 1]
```diff
@@ -645,6 +645,16 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             // Will only work for positive, 32-bit integers
             return make_unique<parser::Integer>(location, std::to_string(intNode->value.value));
         }
+        case PM_INTERPOLATED_MATCH_LAST_LINE_NODE: { // A test of the last read line with interpolation,
```
Comment (st0012) [inline parser/prism/Translator.cc:648]: `A test`? I don't quite get what this means.

File: test/prism_regression/match_last_line.rb [hunk 2]
```diff
@@ -1,11 +1,16 @@
 # typed: false
 
+if // # testing an empty pattern
+#  ^^ error: Unsupported node type `MatchCurLine`
+  end
```
Comment (st0012) [inline test/prism_regression/match_last_line.rb:5]: Maybe remove the redundant spaces?

```suggestion
end
```

**Review by st0012 — APPROVED** [review]

File: test/prism_regression/match_last_line.rb (hunk 2 above)
Comment (amomchilov) [reply test/prism_regression/match_last_line.rb:5]: That's an accident, thanks

File: parser/prism/Translator.cc (hunk 1 above)
Comment (st0012) [reply parser/prism/Translator.cc:648]: Thanks for the context, and WOW.
Maybe something like: `An interpolated regex in a conditional that checks last read line from IO`?

Comment (egiurleo) [reply parser/prism/Translator.cc:648]: I like Stan's description!

**Review by egiurleo — APPROVED** [review]

File: parser/prism/Translator.cc (hunk 1 above)
Comment (amomchilov) [reply parser/prism/Translator.cc]: I had to break it on 2 line to pass the formatter. That gave me some extra room, so I added some more detail:

```suggestion
        case PM_INTERPOLATED_MATCH_LAST_LINE_NODE: { // An interpolated regex literal in a conditional...
            // ...that implicitly checks against the last read line by an IO object, e.g. `if /wat #{123}/`
```

**Pull request merged**